lib/std/start.zig

```diff
@@ -214,10 +216,22 @@ fn EfiMain(handle: uefi.Handle, system_table: *uefi.tables.SystemTable) callconv
         usize => {
             return root.main();
         },
-        uefi.Status => {
+        Status => {
             return @intFromEnum(root.main());
         },
-        else => @compileError("expected return type of main to be 'void', 'noreturn', 'usize', or 'std.os.uefi.Status'"),
+        Status.Error!void => {
+            root.main() catch |err| {
+                // if (err == error.Unexpected) {
+                //     unreachable;
+                // }
+
+                const status = Status.fromError(err);
+                return status;
+            };
+
+            return 0;
+        },
+        else => @compileError("expected return type of main to be 'void', 'noreturn', 'usize', 'std.os.uefi.Status', or 'std.os.uefi.Status.Error!void"),
```

Comment:
I think we should get rid of `usize`, there is no sensible reason why anyone would need it. More type safety is nice.